src/linux/system.rs

```diff
@@ -855,17 +855,10 @@ fn copy_from_file(entry: &Path) -> Vec<String> {
 
 fn get_all_data_from_file(file: &mut File, size: usize) -> io::Result<String> {
     use std::io::Seek;
-
-    let mut data = Vec::with_capacity(size);
-    unsafe {
-        data.set_len(size);
-    }
-
+    let mut buf = String::with_capacity(size);
     file.seek(::std::io::SeekFrom::Start(0))?;
-    let size = file.read(&mut data)?;
-    data.truncate(size);
-    Ok(String::from_utf8(data)
-        .map_err(|e| io::Error::new(io::ErrorKind::InvalidInput, e.to_string()))?)
+    file.read_to_string(&mut buf)?;
```

Comment:
Can't you just return what this function returns?

```rust
file.read_to_string(&mut buf)
```

Or is it a conversion performed by `?`?

Reply:
`file.read_to_string` returns a `Result<usize>`

Reply:
Thanks for the explanation!